dev/benchmarks.el

```diff
@@ -0,0 +1,140 @@
+(require 'dash)
+
+
+(defvar dash-benchmarks nil
+  "List of all defined benchmarks, last declared first.")
+
+(defmacro run-benchmark (num-tries &optional repetitions &rest forms)
+  `(-let ((all-results nil))
+     (--dotimes ,num-tries
+       (garbage-collect)
+       (!cons (benchmark-run-compiled ,repetitions ,@forms) all-results))
+     all-results))
+
+(defmacro defbenchmark (name details &optional repetitions &rest forms)
+  (declare (indent 3))
+  `(!cons (list (quote ,name) (quote ,details) (lambda (num-tries) (run-benchmark num-tries ,repetitions ,@forms)))
+          dash-benchmarks))
+
+(defun should-select-benchmark (name details selector)
+  (cond ((eq selector t)
+         t)
+        ((eq selector nil)
+         nil)
+        ((stringp selector)
+         (string-match selector (format "%s %s" name (or details ""))))
+        ((and (consp selector) (eq (car selector) 'not))
+         (not (should-select-benchmark name details (nth 1 selector))))
+        ((and (consp selector) (eq (car selector) 'and))
+         (--all? (should-select-benchmark name details it) (cdr selector)))
+        ((and (consp selector) (eq (car selector) 'or))
+         (--any? (should-select-benchmark name details it) (cdr selector)))
+        (t
+         (error "Unhandled selector '%s'" selector))))
+
+(defun select-benchmarks (selector)
+  (nreverse (--filter (-let (((name details _ &as benchmark) it))
```

Comment:
You are right. It "sort of works" only because I never use the `benchmark` variable, but it is wrong.  I will commit a fix.